app/src/main/java/pl/ppiwd/exerciseanalyst/common/session/DeviceConnectionChecker.java

```diff
@@ -0,0 +1,66 @@
+package pl.ppiwd.exerciseanalyst.common.session;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.util.Log;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.localbroadcastmanager.content.LocalBroadcastManager;
+
+import pl.ppiwd.exerciseanalyst.activities.utils.MetaMotionServiceConnection;
+import pl.ppiwd.exerciseanalyst.common.BroadcastMsgs;
+
+public class DeviceConnectionChecker {
+
+    private final LocalBroadcastManager broadcastManager;
+    private boolean isDeviceConnected;
+    private final BroadcastReceiver broadcastReceiver;
+    private MetaMotionServiceConnection metaMotionServiceConnection;
+
+    public DeviceConnectionChecker(AppCompatActivity activityClass, MetaMotionServiceConnection metaMotionServiceConnection) {
+        this.metaMotionServiceConnection = metaMotionServiceConnection;
+        this.broadcastReceiver = new BroadcastReceiver() {
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                final String action = intent.getAction();
+                Log.i("DataCollectionActivity", "Received msg with action: " + action);
+                switch (action) {
+                    case BroadcastMsgs.METAMOTION_SERVICE_ALIVE_RESP:
+                        isDeviceConnected = true;
+                        break;
+                    case BroadcastMsgs.ACTIVITY_BROADCAST_INTENT_ACTION:
+                        handleMsgFromService(intent.getStringExtra(BroadcastMsgs.MSG_KEY));
+                        break;
+                }
+            }
+        };
+        this.broadcastManager = LocalBroadcastManager.getInstance(activityClass);
+        this.broadcastManager.registerReceiver(broadcastReceiver, getIntentFilter());
+    }
+
+    public boolean isConnected() {
```

Comment:
Previously it was called isServiceRunning, it is not the same as isConnected. The foreground service might be running independently.